docs/src/pages/backend-adapters/fetch.mdx

```diff
@@ -0,0 +1,145 @@
+import { Callout, Steps } from "nextra-theme-docs";
+
+# Getting started with Fetch / Edge Runtimes
+
+UploadThing is compatible with any runtime that follow the
+[WinterCG](https://wintercg.org/).
+
+## Common setup
+
+### Package Setup
+
+<Steps>
+
+### Install the package
+
+import { Tab, Tabs } from "nextra-theme-docs";
+
+```sh npm2yarn
+npm install uploadthing
+```
+
+#### Add env variables
+
+<Callout>
+  If you don't already have a uploadthing secret key, [sign
+  up](https://uploadthing.com/sign-in) and create one from the
+  [dashboard!](https://uploadthing.com/dashboard)
+</Callout>
+
+```bash copy
+UPLOADTHING_SECRET=... # A secret key for your app (starts with sk_live_)
+UPLOADTHING_APP_ID=... # Your app id
+```
+
+### Set Up A FileRouter
+
+All files uploaded to uploadthing are associated with a FileRoute. Following
+example is very minimalistic. To get full insight into what you can do with the
+FileRoutes, please refer to the
+[File Router API](/api-reference/server#file-routes).
+
+```ts copy filename="uploadthing.ts"
+import { createUploadthing, type FileRouter } from "uploadthing/server";
+
+const f = createUploadthing();
+
+export const uploadRouter = {
+  videoAndImage: f({
+    image: {
+      maxFileSize: "4MB",
+      maxFileCount: 4,
+    },
+    video: {
+      maxFileSize: "16MB",
+    },
+  }).onUploadComplete((data) => {
+    console.log("upload completed", data);
+  }),
+} satisfies FileRouter;
+
+export type OurFileRouter = typeof uploadRouter;
+```
+
+</Steps>
+
+## Runtimes-specific setup
+
+### Astro
+
+```ts filename="pages/api/uploadthing.ts"
+import { createServerHandler } from "uploadthing/server";
+
+import { uploadRouter } from "../../server/uploadthing";
+
+export const { GET, POST } = createServerHandler({
+  router: uploadRouter,
+  config: {
+    uploadthingId: import.meta.env.UPLOADTHING_APPID,
+    uploadthingSecret: import.meta.env.UPLOADTHING_SECRET,
+    callbackUrl: "http://localhost:4321/api/uploadthing",
+  },
+});
+```
+
+### Elysia
+
+```ts filename="src/index.ts"
+import { Elysia } from "elysia";
+import { uploadRouter } from "uploadthing.ts";
+
+import { createServerHandler } from "uploadthing/server";
+
+const { GET, POST } = createServerHandler({
+  router: uploadRouter,
+});
+
+const app = new Elysia().get("/", () => "Hello Elysia");
+
+app.group("/api/uploadthing", (app) =>
+  app
+    .post("/", (context) => {
+      return POST(context.request);
+    })
+    .get("/", (context) => {
+      return GET(context.request);
+    }),
+);
+
+app.listen(3000);
+```
+
+### Hono
+
+```ts filename="src/index.ts"
+import { Hono } from "hono";
+
+import { createServerHandler } from "uploadthing/server";
+
+import { uploadRouter } from "./uploadthing.ts";
+
+const { GET, POST } = createServerHandler({
+  router: uploadRouter,
+});
+
+const app = new Hono();
+const ut = new Hono();
+
+ut.get("/", (c) => {
+  return GET(c.req.raw);
+}).post("/", (c) => {
+  return POST(c.req.raw);
+});
```

Comment:
Same here, just add the routes individually? 

I have never used Hono but seems odd to create a new instance?

Reply:
In case of both Hono and Elysia I just decided to make it through grouping so that the same endpoint "/api/uploadthing" is not repeated twice

In Hono you create new instance of Hono class when you want to create something similar to Express router